tests/cnf/core/network/metallb/tests/bgp-remote-as-dynamic.go

```diff
@@ -0,0 +1,227 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/deployment"
+	"github.com/openshift-kni/eco-goinfra/pkg/metallb"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+	"github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/mlbtypesv1beta2"
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/netinittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/internal/netparam"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/frr"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/metallbenv"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/tsparams"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("BGP remote-dynamicAS", Ordered, Label(tsparams.LabelDynamicREmoteASTestCases),
+	ContinueOnFailure, func() {
+		var (
+			err           error
+			dynamicASiBGP = "internal"
+			dynamicASeBGP = "external"
+			iBGPASN       = 64500
+			eBGPASN       = 64501
+		)
+
+		BeforeAll(func() {
+			By("Getting MetalLb load balancer ip addresses")
+			ipv4metalLbIPList, ipv6metalLbIPList, err = metallbenv.GetMetalLbIPByIPStack()
+			Expect(err).ToNot(HaveOccurred(), tsparams.MlbAddressListError)
+
+			By("List CNF worker nodes in cluster")
+			cnfWorkerNodeList, err = nodes.List(APIClient,
+				metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+
+			By("Selecting worker node for BGP tests")
+			workerLabelMap, workerNodeList = setWorkerNodeListAndLabelForBfdTests(cnfWorkerNodeList, metalLbTestsLabel)
+			ipv4NodeAddrList, err = nodes.ListExternalIPv4Networks(
+				APIClient, metav1.ListOptions{LabelSelector: labels.Set(workerLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Failed to collect external nodes ip addresses")
+
+			err = metallbenv.IsEnvVarMetalLbIPinNodeExtNetRange(ipv4NodeAddrList, ipv4metalLbIPList, nil)
+			Expect(err).ToNot(HaveOccurred(), "Failed to validate metalLb exported ip address")
+
+			By("Listing master nodes")
+			masterNodeList, err = nodes.List(APIClient,
+				metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+			Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+				"Failed to detect master nodes")
+		})
+
+		AfterAll(func() {
+			if len(cnfWorkerNodeList) > 2 {
+				By("Remove custom metallb test label from nodes")
+				removeNodeLabel(workerNodeList, metalLbTestsLabel)
+			}
+		})
```

Comment:
include one more step in AfterAll to remove the metallb test label on worker nodes.